src/Symfony/Component/Translation/Bridge/Phrase/PhraseProvider.php

```diff
@@ -0,0 +1,263 @@
+    }
+
+    public function write(TranslatorBagInterface $translatorBag): void
+    {
+        \assert($translatorBag instanceof TranslatorBag);
+
+        foreach ($translatorBag->getCatalogues() as $catalogue) {
+            foreach ($catalogue->getDomains() as $domain) {
+                if (0 === \count($catalogue->all($domain))) {
+                    continue;
+                }
+
+                $phraseLocale = $this->getLocale($catalogue->getLocale());
+
+                $content = $this->xliffFileDumper->formatCatalogue($catalogue, $domain, ['default_locale' => $this->defaultLocale]);
+                $filename = sprintf('%d-%s-%s.xlf', date('YmdHis'), $domain, $catalogue->getLocale());
+
+                $fields = array_merge($this->writeConfig->setTag($domain)->setLocale($phraseLocale)->getOptions(), ['file' => new DataPart($content, $filename, 'application/xml')]);
+
+                $formData = new FormDataPart($fields);
+
+                $response = $this->httpClient->request('POST', 'uploads', [
+                    'body' => $formData->bodyToIterable(),
+                    'headers' => $formData->getPreparedHeaders()->toArray(),
+                ]);
+
+                if (201 !== $statusCode = $response->getStatusCode()) {
+                    $this->logger->error(sprintf('Unable to upload translations for domain "%s" to phrase: "%s".', $domain, $response->getContent(false)));
+
+                    $this->throwProviderException($statusCode, $response, 'Unable to upload translations to phrase.');
+                }
+            }
+        }
+    }
+
+    public function delete(TranslatorBagInterface $translatorBag): void
+    {
+        $keys = [[]];
+
+        foreach ($translatorBag->getCatalogues() as $catalogue) {
+            foreach ($catalogue->getDomains() as $domain) {
+                $keys[] = array_keys($catalogue->all($domain));
+            }
+        }
+
+        $keys = array_unique(array_merge(...$keys));
+        $names = array_map(static fn ($v): ?string => preg_replace('/([\s:,])/', '\\\\\\\\$1', $v), $keys);
+
+        foreach ($names as $name) {
+            $response = $this->httpClient->request('DELETE', 'keys', [
+                'query' => [
+                    'q' => 'name:'.$name,
+                ],
+            ]);
+
+            if (200 !== $statusCode = $response->getStatusCode()) {
+                $this->logger->error(sprintf('Unable to delete key "%s" in phrase: "%s".', $name, $response->getContent(false)));
+
+                $this->throwProviderException($statusCode, $response, 'Unable to delete key in phrase.');
+            }
+        }
+    }
+
+    private function getCache(CacheItemInterface $item): ?array
+    {
+        if (!$item->isHit() || null === $cache = $item->get()) {
+            return null;
+        }
+
+        return $cache;
+    }
+
+    private function key(string $locale, string $domain, array $options): string
+    {
+        array_multisort($options);
```

Comment:
Why not using `sort` here?

Reply:
some of the options (e.g. `format_options`) can be an array